test/gateway/eip20_gateway/progress_stake_with_proof.js

```diff
@@ -0,0 +1,520 @@
+
+  let setStorageRoot = async function() {
+
+    let blockNumber = new BN(proofData.co_gateway.confirm_stake_intent.proof_data.block_number);
+    let storageRoot = proofData.co_gateway.confirm_stake_intent.proof_data.storageHash;
+    await gateway.setStorageRoot(blockNumber, storageRoot);
+
+    blockNumber = new BN(proofData.co_gateway.progress_mint.proof_data.block_number);
+    storageRoot = proofData.co_gateway.progress_mint.proof_data.storageHash;
+    await gateway.setStorageRoot(blockNumber, storageRoot);
+
+  };
+
+  beforeEach(async function () {
+    mockToken = await MockToken.new({ from: accounts[0] });
+    baseToken = await MockToken.new({ from: accounts[0] });
+
+    let owner = accounts[2];
+    let worker = accounts[7];
+    let organization = await MockOrganization.new(
+      owner,
+      worker,
+      { from: accounts[0] },
+    );
+
+    let coreAddress = accounts[5];
+    let burner = NullAddress;
+
+    bountyAmount = new BN(proofData.gateway.constructor.bounty);
+
+    gateway = await Gateway.new(
+      mockToken.address,
+      baseToken.address,
+      coreAddress,
+      bountyAmount,
+      organization.address,
+      burner,
+    );
+
+    let gatewayLib = await GatewayLib.deployed();
+
+    let stakeParams = proofData.gateway.stake.params;
+
+    stakeData = {
+      amount: new BN(stakeParams.amount, 16),
+      beneficiary: stakeParams.beneficiary,
+      gasPrice: new BN(stakeParams.gasPrice, 16),
+      gasLimit: new BN(stakeParams.gasLimit, 16),
+      nonce: new BN(stakeParams.nonce, 16),
+      hashLock: stakeParams.hashLock,
+      messageHash: proofData.gateway.stake.return_value.returned_value.messageHash_,
+      staker: stakeParams.staker,
+    };
+
+    let stakeIntentHash = await gatewayLib.hashStakeIntent(
+      stakeData.amount,
+      stakeData.beneficiary,
+      proofData.contracts.gateway,
+    );
+
+    stakeData.intentHash = stakeIntentHash;
+
+    await gateway.setStake(
+      stakeData.messageHash,
+      stakeData.beneficiary,
+      stakeData.amount,
+    );
+
+    await gateway.setStakeMessage(
+      stakeData.messageHash,
+      stakeData.intentHash,
+      stakeData.nonce,
+      stakeData.gasPrice,
+      stakeData.gasLimit,
+      stakeData.staker,
+      stakeData.hashLock
+    );
+
+    await gateway.setOutboxStatus(
+      stakeData.messageHash,
+      MessageStatusEnum.Declared
+    );
+
+    progressStakeParams = {
+      messageHash: stakeData.messageHash,
+      rlpParentNodes: proofData.co_gateway.confirm_stake_intent.proof_data.storageProof[0].serializedProof,
+      blockHeight: new BN(proofData.co_gateway.confirm_stake_intent.proof_data.block_number),
+      messageStatus: MessageStatusEnum.Declared,
+    };
+
+    await mockToken.transfer(gateway.address, stakeData.amount, { from: accounts[0] });
+    await baseToken.transfer(gateway.address, bountyAmount, { from: accounts[0] });
+
+  });
```

Comment:
That is a very long `beforeEach`. Can it be split into functions (that are declared at the bottom)?